Sources/Common/Extensions/StringExtension.swift

```diff
@@ -93,6 +95,235 @@ public extension String {
+
+        // next find all non-matched expressions looking like a file path
+        // 1. find `/something` pattern
+        for match in FileRegex.filePathStart.matches(in: self, range: fullRange) {
+            guard let matchIndices = Range(match.range), let matchRange = Range(match.range, in: self),
+                  !result.intersects(integersIn: matchIndices) else { continue /* already matched */ }
+
+            // 2. look backwards for possibly relative path first component start (limited by a whitespace or newline)
+            var pathStartIdx = self.rangeOfCharacter(from: .whitespacesAndNewlines, options: .backwards, range: startIndex..<matchRange.lowerBound)?.upperBound ?? startIndex
+            // 3. look backwards for a possible URL scheme
+            if let schemeRange = self.firstMatch(of: FileRegex.urlScheme, range: startIndex..<pathStartIdx)?.range(in: self) {
+                pathStartIdx = schemeRange.lowerBound
+            }
+            // 4. heuristically find the end of the path
+            let pathEndIdx = self.findFilePathEnd(from: pathStartIdx)
+
+            var resultRange = pathStartIdx..<pathEndIdx
+            dropLineNumberAndTrimSpaces(&resultRange)
+
+            guard let pathRange = Range(NSRange(resultRange, in: self)), pathRange.count > 2 else { continue }
+            // collect the result
+            result.insert(integersIn: pathRange)
+        }
+
+        // next find all non-matched expressions looking like a file name (filename.ext)
+        for match in FileRegex.fileName.matches(in: self, range: fullRange) {
+            guard let matchIndices = Range(match.range(at: 1)), matchIndices.count > 2, var matchRange = Range(match.range, in: self),
+                  !result.intersects(integersIn: matchIndices) else { continue /* already matched */ }
+
+            dropLineNumberAndTrimSpaces(&matchRange)
+
+            guard let pathRange = Range(NSRange(matchRange, in: self)), pathRange.count > 2 else { continue }
+            // collect the result
+            result.insert(integersIn: pathRange)
+        }
+
+        return result.rangeView.compactMap {
+            guard let range = Range(NSRange($0), in: self) else {
+                assertionFailure("Could not convert \($0) to Range in \(self)")
+                return nil
+            }
+            return range
+        }
+    }
+
+    private func findFilePathEnd(from pathStartIdx: String.Index) -> String.Index {
+        // macOS file names can contain literally any Unicode character except `/` and newline with max length=255
+        // but let‘s assume some general naming conventions:
+        // - `filename.extension` followed by a word boundary [ ,.:;], not `/`, terminates the file path
+        // - file/folder names should not contain trailing spaces
+        //   although technically it‘s possible, but if the next path component starts with `/` we'll treat it as another path
+        //
+        // 1. find end of the line
```

Comment:
I winder if we are a bit generous here, e.g. `"Error in /var/filename.txt In file included from /Volumes/data/framework dir".sanitized()` returns `Error in <removed>` 

I'd love to see more tests for simple strings that showcase what is expected to match and what not, with a bit clearer test names :)